### ISSUE TITLE
ISIS SANS Backend rebinning event workspaces

### ISSUE THREAD

**Issue by Mantid-Matthew** [mantidproject/mantid#23973]
When reducing event data the new ISIS Sans backend keeps the data in event mode for as long as possible to maximize speed and accuracy. This is in contrast to the old backend which re-binned it to histogram data as one of the first steps. 

This is causing an issue when masking bins is applied. When MaskBins is applied to an event workspace the masked events are removed and no record is kept that the masking was done. When the normalization in Q1D then tries to take account of bin masking in the normalization it thinks no bins have been masked when they have causing it to over normalize a certain Q range leading to a noticeable dip feature in the output.

The suggested fix for this is to reorder the reduction workflow such that Bin Masking is always done after the workspace has been rebinned into a histogram workspace. In practise this involves moveing the bin masking to after step 9 here 

https://github.com/mantidproject/mantid/blob/master/Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SANS/SANSReductionCore.py#L206

The general consensus is that this should not change the reduction outcomes but this should be verified particularly in the case where wide angle correction are used as it may the case that the bin masking has to done before the adjustment workspaces are calculated in this case. 

The bin masking is currently done by SANSMask.py the normalisation error is occurring here

https://github.com/mantidproject/mantid/blob/master/Framework/Algorithms/src/Q1D2.cpp#L582

**Comment by Mantid-Matthew** [comment]
Having talked to @OwenArnold it was suggested that fixing how event lists handle masking so that they do preserve the fact they have been masked would be a better fix here. This would involve changing the core handling of the event work spaces in a way I need to investigate.

**Comment by TTitcombe** [comment]
Re-ordering SANS reduction workflow is not (easily) possible.
Disabling compatibility mode to avoid users from experience innacurate data #24162 
New issue to fix is with Rebin algorithm #24092

**Comment by TTitcombe** [comment]
The solution is to create a histogram workspace copy of the workspace, provide this copy with mask bins and perform all rebin operations on it as you would the main workspace. After converting the main workspace to histogram, copy the bins over
